addons/im_livechat/static/src/models/chat_window/chat_window.js

```diff
@@ -0,0 +1,22 @@
+odoo.define('im_livechat/static/src/models/chat_window/chat_window.js', function (require) {
+'use strict';
+
+const {
+    registerInstancePatchModel,
+} = require('mail/static/src/model/model_core.js');
+
+registerInstancePatchModel('mail.chat_window', 'im_livechat/static/src/models/chat_window/chat_window.js', {
+
+    /**
+     * @override
+     */
+    close({ notifyServer = true } = {}) {
+        const thread = this.thread;
+        if (thread && thread['channel_type'] === 'livechat' && !thread['messages'].length) {
```

Comment:
Yes checking `isLoaded` should be enough, but it should be checked on the `mainCache` instead of the arbitrary `caches[0]` ([0] should usually be the maincache but we never know).